corehq/form_processor/models.py

```diff
@@ -456,10 +457,15 @@ def write_content(self, content):
             content_readable = StringIO(content)
         elif isinstance(content, six.binary_type):
             content_readable = BytesIO(content)
-
         db = get_blob_db()
         bucket = self.blobdb_bucket()
-        info = db.put(content_readable, get_short_identifier(), bucket=bucket)
+        if self.blob_id:
+            # Delete the old entry in the database for this identifier
+            db.delete(self.blob_id, bucket)
```

Comment:
I don't think it's necessary to delete first. `put` should overwrite an existing blob (if any). Because of the distributed and eventually consistent nature of Riak CS (or other S3 implementations) it take a sort time (probably on the order of seconds or less) for subsequent reads to get the new value. Deleting first will not change the delayed consistency, FWIW.